docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,527 @@
+* Populating the listing with data obtained from the napari hub API [^napari-hub-api].
+* Updating the listing to filter and mark plugins based on their availability on conda-forge.
+
+Some UX/UI improvements that are not part of this milestone but could be implemented in future
+releases include:
+
+* Making the plugin dialog a side panel / side dockwidget (similar to VS Code
+  [^vscode-extensions-ui])
+* Allow for simultaneous install of plugins. Currently multiple plugins can be selected for
+  install, uninstall and update, but each on of these actions, are queued and run sequentially.
+* Allow advanced users to install packages from PyPI or other local sources via drag-and-drop. This
+  presents many risks that could irreversibly disrupt the installation, though, so it needs to be
+  considered very carefully.
+
+### Milestone 4: Enabling in-app napari version updates
+
+Our first attempts [^in-app-update-pr] to implementing this consisted of a pop-up dialog that would
+notify the user about the available updates. Accepting the dialog would trigger the creation of a
+new environment (e.g. `napari-0.4.16`) alongside the existing one (e.g. `napari-0.4.15`). While it
+could be argued that this is not an "update" per se, but a new installation, it fulfills the same
+goal: the new version is available. If the user wants to get rid of the old one, an option would
+have been offered to auto-clean on success, or after not using it for some configurable time.
+
+However, having several versions of napari coexisting might risk questions like "which napari
+version is the default one?", "which one will be in charge of updating napari?", "what if the
+update logic changes over time?". There possible workarounds, but from the user perspective it could
+get confusing very fast.
+
+For that reason, we decided to slightly redesign our approach to in-app updates. We will still use
+several environments to keep things tidy and performant (see "Detailed description" for Milestone
+2 for more details), but the update checks will mimic something closer to a a server-client model:
+
+* The server: We will create a separate standalone package to handle updates outside of the
+  `napari` process. This will be designed in an application agnostic way while targeting the needs
+  of the napari project. It will allow the user to handle updates but also manage the existing
+  napari installations (investigate issues, remove old ones, clean expired caches, etc).
+* The client: each napari version will have a very basic notion about the "update server" and will
+  query for updates every time it starts, and also periodically while running. If one is found, it
+  the server will run the update on its own.
+
+The governance of the "server" part will still fall under the `napari` organization (much like
+`magicgui`) but it will be usable outside of the napari project, following the same philosophy we
+adopted for the constructor work.
```

Comment:
I don't really understand these, could you elaborate here? Is this a *literal* server model (napari will ping not conda-forge but a new application), or are you just saying that there are two applications locally (like e.g. Microsoft Autoupdater or whatever it is)? Perhaps references to other software that works this way will help clarify things for me.

Reply:
Yes, let me reword this part. It doesn't read as clear as I intended :D It's not an online server at all, but more like a service that can serve requests to the client.

For example: a running `napari` instance  will ask `napari-updater` (maybe a running service, maybe a one-shot executable running in a separate process) whether there's a new napari version or not. If there's one, it will install it on its own (`napari` knows nothing about how to do that), on its own separate process.

I'll cover this in the next commit.

Reply:
Ready now!